### ISSUE TITLE
Piped not playing videos

### ISSUE THREAD

**Issue by probatschool2** [TeamPiped/Piped#2809]
### Official Instance

- [ ] The bug is reproducible on the [official hosted instance](http://piped.video/), or is API-related.

### Describe the bug

For some reason the videos dont load and I cant watch piped. I primarily use piped cause youtube is blocked and this piped link is just about one of the only ones not blocked. If someone can please help me with this it would be much appriciated! ![Screenshot 2023-08-14 11 17 01 PM](https://github.com/TeamPiped/Piped/assets/142287487/29ab56f3-0eb0-4c0b-bac7-8cbfd210112c)

### To Reproduce

i honestly dont know how to reproduce this bug

### Expected behavior

I expected to be able to watch the youtube videos I want to.

### Browser, and OS with Version.

I use google and i think I may use chrome os beacuse Im on my school chrome book and use it.

**Comment by alejandro-alzate** [comment]
same issue here; mine is reproducible on the lastest libretube client of fdroid with the instance `piped.yt` my suspicious is that instances are being bot-blocked correct me if the case.
Unrelated to the bug, how can i run an instance for my people if theres an article for this please link it to me.

**Comment by DarkCat09** [comment]
The same issue with my `ytapi.dc09.ru` instance.

Proxy returns 403 Forbidden on every video stream request. I mean previews and channel avatars are succesfully loaded, but URLs from videoStreams and audioStreams are not.

I'll update API and proxy, and then edit this comment if anything changed.

**Comment by DarkCat09** [comment]
Found the same issue in Invidious repo, and it have been already fixed: https://github.com/iv-org/invidious/issues/4045#issuecomment-1674373088

**Comment by DarkCat09** [comment]
Fix in NewPipeExtractor: https://github.com/TeamNewPipe/NewPipeExtractor/pull/1084

**Comment by DarkCat09** [comment]
This bug still exists even on the latest Piped-Backend commit.

**Comment by fahim44** [comment]
no instance work right now. For example: GET https://watchproxy-nl.whatever.social/videoplayback returns 403.

**Comment by probatschool2** [comment]
I dont know all these terms yall use like instances and stuff like that i m only in 9th grade just trying to find a unblocked youtube to use

**Comment by Qaz-6** [comment]
ReVanced makes patches in secret. When some people share APKs, YouTube takes them down. ReVanced patches are a good idea, because Google doesn't understand how apks are made. Google makes fun of Piped, but it doesn't take Piped seriously

**Comment by Fntoomen** [comment]
> ReVanced makes patches in secret. When some people share APKs, YouTube takes them down. ReVanced patches are a good idea, because Google doesn't understand how apks are made. Google makes fun of Piped, but it doesn't take Piped seriously

🤣 I see we have some ReVanced dev here.

**Comment by Qaz-6** [comment]
https://api.piped.projectsegfau.lt
It works

**Comment by probatschool2** [comment]
sadly the website https://api.piped.projectsegfau.lt/ is blocked and not working for me they have it blocked by category on my computer.

**Comment by FireMasterK** [comment]
> @FireMasterK, could you update your Extractor fork? [TeamNewPipe/NewPipeExtractor#1084](https://github.com/TeamNewPipe/NewPipeExtractor/pull/1084)

This is unrelated and we already have this commit.

**Comment by Qaz-6** [comment]
Now all instances are working

**Comment by smol-loli** [comment]
Cannot confirm this. Tested the offical, piped.adminforge.de and piped.smnz.de and they all don't work.

**Comment by alejandro-alzate** [comment]
instance `whatever.social` broke, i migrated to [REDACTED] but broke as well WHILE writing this down, migrated to another is the piped network being attacked?

**Comment by lmenale** [comment]
Still doesn't work

**Comment by Dominion0815** [comment]
yes still broken. I think google blocked again. piped.adminforge.de is now running with privoxy/wireguard vpn and it works but the performance is realy bad.

**Comment by DarkCat09** [comment]
[piped.mha.fi](https://piped.mha.fi) and [project segfault](https://piped.projectsegfau.lt) work.

I don't know how, but if Google actually banned proxy IPs, I can request other address from my hoster for dc09.ru instance. Will it help?

**Comment by n-ce** [comment]
Albeit slighly unrelated but a collaboration on manually maintaining a working instances list is urgently needed. Is anyone interested in leading such a project?

**Comment by BlackLight** [comment]
@n-ce if Google is straight away banning IP addresses of ytproxy servers is it really worth? It may just be a matter of time before those servers get banned too, if they use static addresses. Maybe it'd be more helpful to compile a list of instances that will start adopting rotating IP addresses, if it really turns out that Google is blacklisting requests on an IP basis.

**Comment by giselle-campos** [comment]
This is informative but unfortunate...

**Comment by gamer191** [comment]
> YouTube has blocked access to Piped proxy from connecting to YouTube

Have they though? Do youtube videos work in a browser on the same server, or is there something more to it?

**Comment by n-ce** [comment]
Issue seems to be fixed.
Tested on :
- piped.video
- garudalinux.org
- privacydev.net
- adminforge.de

**Comment by d4rklynk** [comment]
Hi, is changing the server IP the only solution rn ?

**Comment by r1klz** [comment]
> Issue seems to be fixed. Tested on :
> 
>     * piped.video
> 
>     * garudalinux.org (edit : [failing with authorisation errors](https://piped-proxy.garudalinux.org/videoplayback?expire=1692527517&ei=PZfhZICQN7KH6dsPsayZmA4&ip=116.202.208.112&id=o-AE9-zwDnCVqb7n7bf_8dcOorKLOYAE7VnPGOkWTzZSfe&itag=249&source=youtube&requiressl=yes&mh=BW&mm=31%2C26&mn=sn-4g5ednz7%2Csn-h0jeenl6&ms=au%2Conr&mv=m&mvi=4&pl=17&initcwndbps=520000&spc=UWF9f_N1Jb5brb9vJHaIqmRVXlew_n8&vprv=1&svpuc=1&mime=audio%2Fwebm&gir=yes&clen=2704356&dur=408.381&lmt=1574756811747966&mt=1692505501&fvip=3&keepalive=yes&fexp=24007246%2C24362685%2C51000023&beids=24350017&c=ANDROID&txp=5531432&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgZiP4wUo7xEbUpqPbCgYNV5rzkOQz0BXBf7c3xxRj2AMCIQCrdEIg_Bu2bv_oKtUAWmZYKGpk9wH0aR8j-BrMRnRnFw%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgPaoRwPCLvFKbHArgyzRJnFNk1VQPUQB5D9-eKoV8AEICIQCwThDeP1iNTpUto6-oYIxSLPua-sQ0S6zyFgy_K5hyeA%3D%3D&cpn=PVKoetfzkY2YJIOh&host=rr4---sn-4g5ednz7.googlevideo.com))
> 
>     * privacydev.net
> 
>     * adminforge.de

Moreover
- kavin.rocks libre
- mha.fi
- qdi.fi

proxies;
are working too!

**Comment by drgns-wtd** [comment]
Piped proxy just stoped working for drgns.space :( I have a static IP so I'm not sure how to get my IP clean again, but for now I put piped-proxy behind a VPN to get it working :+1:

**Comment by FireMasterK** [comment]
> Hi, is changing the server IP the only solution rn ?

You can use a VPN. That's what the official instance currently does.

**Comment by d4rklynk** [comment]
I can install ProtonVPN on my instance if needed yeah. But it always will be the same IP, I'm not sure I can change servers.

**Comment by Moonstone3010** [comment]
You can change servers, but there are a limited number of them. I suggest you remove the name of VPN provider from this message just in case :)

**Comment by drgns-wtd** [comment]
> Piped proxy just stoped working for drgns.space :( For now I put piped-proxy behind a VPN to get it working 👍

> yes still broken. I think google blocked again. piped.adminforge.de is now running with privoxy/wireguard vpn and it works but the performance is realy bad.

I'm getting really bad performance too. 
Using the same VPN on my device and disabling the proxy works flawless though, I'm not sure why

**Comment by d4rklynk** [comment]
If a VPN is really the solution in the long run, it's not good. Hope you can fix it somehow :/

**Comment by DarkCat09** [comment]
dc09.ru [Piped](https://yt.dc09.ru) and [Invidious](https://inv.dc09.ru) instances work now, I've set up a socks5 proxy.

**Comment by drgns-wtd** [comment]
For instance maintainers who use docker but don't want to put the whole system behind a VPN, you can use gluetun as such: (truncated)
```yml
version: "3"

services:
    vpn:
      image: qmcgaw/gluetun
      cap_add:
        - NET_ADMIN
      environment:
        - VPN_SERVICE_PROVIDER=mullvad
        - VPN_TYPE=openvpn 
        - OPENVPN_USER= # Account code
        - SERVER_COUNTRIES=USA
      privileged: true
      restart: unless-stopped
    piped-proxy:
      image: 1337kavin/piped-proxy:latest
      network_mode: "service:vpn" # IMPORTTANT
      restart: unless-stopped
      environment:
          - UDS=1
      volumes:
          - piped-proxy:/app/socket
  ```

If you use a different provider, gluetun supports anything including custom configs:
https://github.com/qdm12/gluetun-wiki/tree/main/setup/providers

**Comment by DarkCat09** [comment]
For instance maintainers who run a VPN/proxy client supporting connection over HTTP or SOCKS (like Xray), you can tell piped-proxy to use it with the env var:
```yaml
piped-proxy:
  ...
  environment:
    PROXY=socks5://127.0.0.1:2080
```

Or in OpenRC config file instead of Docker in my case:
```bash
#!/sbin/openrc-run

export PROXY=socks5://127.0.0.1:2080
...
```

**Comment by d4rklynk** [comment]
> For instance maintainers who use docker but don't want to put the whole system behind a VPN, you can use gluetun as such: (truncated)
> 
> ```yaml
> version: "3"
> 
> services:
>     vpn:
>       image: qmcgaw/gluetun
>       cap_add:
>         - NET_ADMIN
>       environment:
>         - VPN_SERVICE_PROVIDER=mullvad
>         - VPN_TYPE=openvpn 
>         - OPENVPN_USER= # Account code
>         - SERVER_COUNTRIES=USA
>       privileged: true
>       restart: unless-stopped
>     piped-proxy:
>       image: 1337kavin/piped-proxy:latest
>       network_mode: "service:vpn" # IMPORTTANT
>       restart: unless-stopped
>       environment:
>           - UDS=1
>       volumes:
>           - piped-proxy:/app/socket
> ```
> 
> If you use a different provider, gluetun supports anything including custom configs: https://github.com/qdm12/gluetun-wiki/tree/main/setup/providers

Omg man, thanks, it's way easier than setup a VPN.
That's actually a good solution rn.

**Comment by Qaz-6** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/10352#issue-1854180200

**Comment by d4rklynk** [comment]
It works for me though

**Comment by DarkCat09** [comment]
@MuubNito, yt.dc09.ru and piped.projectsegfau.lt work.  
If you meant the official instance piped.video, looks like it overloaded with requests (or maybe it's my ISP issues).

**Comment by hugoamvieira** [comment]
I've also just had this happen today after an update. It's happening in piped.video too when playing a livestream. From the looks of it, it's some frontend problem as the proxy is replying with data and also I can watch the livestreams on the phone through Yattee.

**Comment by FireMasterK** [comment]
> I've also just had this happen today after a frontend update. It's happening in piped.video too when playing a livestream. From the looks of it, it's some frontend problem as the proxy is replying with data but no playback occurs, and also I can watch the livestreams on the phone through Yattee.
> 
> Update: I built the frontend from source, version `master @ cc478fd5` and it now works. I suspect the shaka player update broke something

This appears to be a different issue, please open a new issue for this! This issue is for the Google 403 issue

**Comment by MuubNito** [comment]
> @MuubNito, yt.dc09.ru and piped.projectsegfau.lt work. If you meant the official instance piped.video, looks like it overloaded with requests (or maybe it's my ISP issues).

It was the official instance, thanks for explaining :)

**Comment by FireMasterK** [comment]
> Any update for this issue ? Do we still need to use VPN or so ?

It looks like you still need to. I checked with of my IPs that was affected, and it still appears to be affected by this issue.

**Comment by Qaz-6** [comment]
I don't use VPN. I closed Piped proxy. Videos open easily.

**Comment by d4rklynk** [comment]
What are the consequences of disabling proxy ? Google will see IP of users ? And what ?

If it's IP, I totally don't care.

But if for example, using Piped w/o proxy or LibreTube with proxy-bypass feature will show my IP users to Google, can they potentially block account that are used from the same IP ?

I had similar problem when using Aurora Store months ago. My account Google has been irrevocably blocked. Aurora does not respect ToS. I don't know if Piped does though.

**Comment by cyberhardt** [comment]
pretty much all instances just take a million years to load videos if they load at all. very frustrating. as I've mentioned in some other issue, NewPipe works much better. What does NewPipe has that the desktop instances don't?

**Comment by absidue** [comment]
@cyberhardt NewPipe does everything on your device and talks directly to YouTube, so your IP address, search queries etc are exposed to them. Piped does everything through the server, including the video streaming, so YouTube has no idea what comes from you and what comes from other users on the same instance. The downside of Piped's approach, is that YouTube is not a fan of the same IP address streaming 100s of terabytes of video and audio every month, so the  do waves of IP bans for streaming, all the API endpoints still work fine.

The only solution, without sacrificing your privacy, is for the server to get a new IP address.

**Comment by cyberhardt** [comment]
> @cyberhardt NewPipe does everything on your device and talks directly to YouTube, so your IP address, search queries etc are exposed to them. Piped does everything through the server, including the video streaming, so YouTube has no idea what comes from you and what comes from other users on the same instance. The downside of Piped's approach, is that YouTube is not a fan of the same IP address streaming 100s of terabytes of video and audio every month, so the do waves of IP bans for streaming, all the API endpoints still work fine.
> 
> The only solution, without sacrificing your privacy, is for the server to get a new IP address.

I see, thanks for the explanation

**Comment by d4rklynk** [comment]
How do you remove proxy part in Piped ?

**Comment by absidue** [comment]
@d4rklynk Setup your own Piped instance without the proxy and modify the code in Piped and LibreTube to not use the proxy.

**Comment by Qaz-6** [comment]
![Screenshot_20231006-143054_LibreTube Debug](https://github.com/TeamPiped/Piped/assets/140435478/d447e249-b5c0-4b20-a36d-8050bfd0c906)
This is written when this video opens.Also some 6 minute videos open in 3 minutes and often freeze or stop.Videos are close to region.Server is kavin.rocks.Server is supported in my country and no region restrictions.Internet connection is great.🤔

**Comment by Qaz-6** [comment]
kotlin.UninitializedPropertyAccessException: lateinit property streams has not been initialized
	at com.github.libretube.ui.fragments.PlayerFragment.initializeOnClickActions$lambda$4(PlayerFragment.kt:404)
	at com.github.libretube.ui.fragments.PlayerFragment.$r8$lambda$x2GIe87d5halYB00x6Jpj3yT0Ko(Unknown Source:0)
	at com.github.libretube.ui.fragments.PlayerFragment$$ExternalSyntheticLambda6.onClick(Unknown Source:2)
	at android.view.View.performClick(View.java:7792)
	at android.view.View.performClickInternal(View.java:7769)
	at android.view.View.access$3800(View.java:910)
	at android.view.View$PerformClick.run(View.java:30218)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8669)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)

**Comment by alex-djangsters** [comment]
I have kinda similar issue, with `piped.video`: video starts playing fine, but after random period of 2 to 8 minutes it stops and infinite loading is happening, the period is different for each video, but always the same for same video. A good example is https://piped.video/watch?v=lMKK7DUVEmQ it freezes somewhere around 8:45. Tried different browsers on different platforms, result is always the same. iPadOS: Safari, Opera, Firefox; Arch Linux: Firefox, Chromium.

**Comment by Real-Gecko** [comment]
I have kinda similar issue, with `piped.video`: video starts playing fine, but after random period of 2 to 8 minutes it stops and infinite loading is happening, the period is different for each video, but always the same for same video. A good example is https://piped.video/watch?v=lMKK7DUVEmQ it freezes somewhere around 8:45. Tried different browsers on different platforms, result is always the same. iPadOS: Safari, Opera, Firefox; Arch Linux: Firefox, Chromium.

**Comment by Moonstone3010** [comment]
> I have kinda similar issue, with `piped.video`: video starts playing fine, but after random period of 2 to 8 minutes it stops and infinite loading is happening, the period is different for each video, but always the same for same video. A good example is https://piped.video/watch?v=lMKK7DUVEmQ it freezes somewhere around 8:45. Tried different browsers on different platforms, result is always the same. iPadOS: Safari, Opera, Firefox; Arch Linux: Firefox, Chromium.

Do you use a VPN?

**Comment by Real-Gecko** [comment]
> > I have kinda similar issue, with `piped.video`: video starts playing fine, but after random period of 2 to 8 minutes it stops and infinite loading is happening, the period is different for each video, but always the same for same video. A good example is https://piped.video/watch?v=lMKK7DUVEmQ it freezes somewhere around 8:45. Tried different browsers on different platforms, result is always the same. iPadOS: Safari, Opera, Firefox; Arch Linux: Firefox, Chromium.
> 
> Do you use a VPN?

Nope, but I'm located in Kazakhstan, just in case.

**Comment by Moonstone3010** [comment]
I have the same problem. Longer videos stop at the eighth minute of playback. I have to press Ctrl+F5 to continue playing and from then on the video stops about every two minutes until the end of playback.

**Comment by np22-jpg** [comment]
If this is an IP banning issue, then maybe [Invidious’s fix](https://docs.invidious.io/ipv6-rotator/#if-you-are-running-invidious-in-docker) could apply, where they appear to force the use of IPv6 addresses for DNS resolution then regularly rotate their instance’s public IPv6 address?
